c7n/resources/ecs.py

```diff
@@ -613,6 +614,49 @@ def process(self, resources):
                         'Message'] != 'The specified task definition does not exist.':
                     raise
 
+@TaskDefinition.action_registry.register('delete-permanently')
+class DeleteTaskDefinitionPermanently(BaseAction):
+    '''Delete a task definition permanently
+    
+    This will work in conjunction with the 'delete' action. As a task definition is deregistered, 
+    it can also be deleted
+    
+    .. code-block:: yaml
+
+       policies:
+         - name: delete-task-definition-permanently
+           resource: ecs-task-definition
+           filters:
+             - family: test-task-def
+           actions:
+             - type: delete
+             - type: delete-permanently
```

Comment:
Hey thanks for the work on this one @timmygrable ! This certainly seems useful, though having `delete` and `delete-permanently` actions feels pretty awkward to me (that's not a dig at you, you're just trying to work around a `delete` action that doesn't actually delete).

I'm wondering if the functionality here would be clearer as an opt-in parameter on the `delete` action itself though. I'd worry about any lag time waiting for the deregister, but that'd also affect chaining 2 actions in the same policy as in this example.

A pattern we see elsewhere is adding something like a `force` parameter which would be false by default. In this case it might be clearer to have something like `deregister` (true by default)  similar to how the `remove-keys` action on an `aws.iam-user` has a `disable` option.

Reply:
Good idea, I will make that change. Thanks!